lib/ddtrace/contrib/faraday/middleware.rb

```diff
@@ -0,0 +1,52 @@
+require 'faraday'
+require 'ddtrace/ext/http'
+require 'ddtrace/ext/net'
+
+module Datadog
+  module Contrib
+    module Faraday
+      # Middleware implements a faraday-middleware for ddtrace instrumentation
+      class Middleware < ::Faraday::Middleware
+        def initialize(*)
+          @tracer = try_pin(:tracer, Datadog.tracer)
+          @service = try_pin(:service, SERVICE)
+
+          super
+        end
+
+        def call(env)
+          tracer.trace(SERVICE) do |span|
+            span.resource = env[:method].to_s.upcase
+            span.service = service
```

Comment:
Two things here:
* `service` is ok to have a global one, though some users may want a service for each (let's say) domain. In most of the cases, the domain break-down should be safe so we may follow this path, but for others may be not (i.e. unusable cardinality). Is it possible to provide a configuration to enable/disable this behavior? Users may still want to set the service name for their calls. On the other hand, I don't want to provide n+1 ways to handle the service name in this integration. What easy options do we have here? As always, I prefer the simplest approach that can be easily iterated and improved with users feedbacks, and I don't want to risk having something broken that we should remove.
* `resource` is very specific and risky since it could have huge cardinality. I'd say to keep it with the method name